projects/mercury/src/file/buttons/CreateDirectoryButton.js

```diff
@@ -1,13 +1,27 @@
-import React, {useState} from 'react';
+import React, {useContext, useState} from 'react';
 import useIsMounted from "react-is-mounted-hook";
 import FileNameDialog from "./FileNameDialog";
 import {useFormField} from "../../common/hooks/UseFormField";
 import {isValidFileName} from "../fileUtils";
+import VocabularyContext from "../../metadata/vocabulary/VocabularyContext";
 
-const CreateDirectoryButton = ({children, disabled, onCreate}) => {
+const CreateDirectoryButton = ({children, disabled, onCreate, parentEntityType}) => {
     const [opened, setOpened] = useState(false);
     const isMounted = useIsMounted();
 
+    const {hierarchy} = useContext(VocabularyContext);
+
+    let allowedTypes = "";
+    if (!hierarchy) {
+        allowedTypes = ["Error"];
```

Comment:
Or maybe just "Directory"? If we don't want to support a case, when hierarchy is not defined, then it's better to just throw an error here.

Reply:
I agree. When working further with entity creation I'll see how we can manage this.

Reply:
A hierarchy should always be present, even if it is just be with Directory as single node. So an exception is valid here.